verso/src/lib.rs

```diff
@@ -336,13 +336,11 @@ impl VersoviewController {
 
     /// Get the window's size
     pub fn get_size(&self) -> Result<PhysicalSize<u32>, Box<ipc_channel::ErrorKind>> {
+        let mut size_response = self.event_listeners.size_response.lock().unwrap();
         self.sender.send(ToVersoMessage::GetSize)?;
         let (sender, receiver) = std::sync::mpsc::channel();
-        self.event_listeners
-            .size_response
-            .lock()
-            .unwrap()
-            .replace(sender);
+        size_response.replace(sender);
```

Comment:
Moving the `self.sender.send(ToVersoMessage::GetSize)?;` after `.replace(sender);` might also work? and we can reduce the locking time as well. But we'll need to handle the `Err` if the sender sends fail.

Reply:
I think this will do for now, I'm planning on making it callable in multiple threads in the future, and the implementation could do this then (the error handling is going to be quite weird right now if we do this, since we need to revert the listener to `None`)